CMake < 3.8.2 does not know c++17.

Currently set minimum is 3.3.
Looks like 3.8.2 does support it: https://cmake.org/cmake/help/v3.8/prop_tgt/CXX_STANDARD.html

As reported by @scinopode:
```
CMake Error in ThirdParty/logog/CMakeLists.txt:
  Target "logog" requires the language dialect "CXX17" , but CMake does not
  know the compile flags to use to enable it.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@
 #####################
 
 # Specify minimum CMake version
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.8)
 
 # Set CMake policies
 cmake_policy(SET CMP0011 NEW)

